sdk/core/core-http/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -66,7 +66,7 @@ export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
   constructor(
     nextPolicy: RequestPolicy,
     options: RequestPolicyOptions,
-    private tokenCache: AccessTokenCache,
+    private tokenCache: ExpiringAccessTokenCache,
```

Comment:
this strong coupling feels like a bad idea

Reply:
Resolving since I reverted this.